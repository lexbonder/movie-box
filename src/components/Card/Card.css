@import url('https://fonts.googleapis.com/css?family=Russo+One');

.cardWrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.Card {
  width: 300px;
  margin: 20px;
}

.Card .posterWrapper {
  background-color: rgba(70, 70, 70, 0.315); 
  position: relative;
  height: 450px;
  width: 300px;
  padding: 50px 10px;
}

.Card img {
  position: absolute;
  top: 0;
  left: 0;
  width: 300px;
  height: 450px;
  opacity: 1;
}

.Card button {
  position: absolute;
  right: -2px;
  top: 0px;
  font-size: 40px;
  color: rgba(142, 142, 142, 0.9);
  background-color: transparent;
  height: 50px;
  width: 50px;
  border: none;
  text-shadow: 2px 2px 2px rgb(0,0,0);
  outline: none;
  cursor: pointer;

  z-index: 3;
}


.Card button:hover,
.Card button.active:hover {
  color: rgb(210, 185, 0);
  text-shadow: 0 0 5px rgb(210,185,0);
}

.Card button:active,
.Card button.active:active {
  text-shadow: 0 0 10px rgb(210,185,0);
}

button.front
{
  right: 5px;
}

button.back{
  left: 5px;
}

.Card button.active{
  color: rgb(210, 185, 0);
  text-shadow: 1px 1px 1px rgba(210, 185, 0, .2);
}

.textBox {
  position: absolute;
  top: 50;
  left: 0;
  overflow: auto;
  height: 380px;
  overflow: auto;

  padding: 0 15px;
}

.flip-container {
  perspective: 1000px;
}

.flip-container:hover .flipper, .flip-container.hover .flipper {
  box-shadow: -5px 5px 10px rgba(0, 0, 0, 0.926);
 transform: rotateY(180deg)
}

.flipper {
  transition: .6s;
  transform-style: preserve-3d;

  position: relative;
}

.front, .back {
  backface-visibility: hidden;
}

.front {
  transform: rotateY(0deg);
}

.back{
  /* background-color: rgb(97, 97, 100); */
  transform: rotateY(180deg);
}

.movie-title {
  color: rgb(170, 0, 0);
  font-family: 'Russo One', sans-serif;
  font-size: 25px;
  font-weight: lighter;
  letter-spacing: 1.5px;
  margin-bottom: -10px;
}

p {
  color: rgba(255, 255, 255, 0.6);
  font-size: 18px;
  text-align: left;
  letter-spacing: 1px;
}

.date {
  font-size: 14px;
  margin-bottom: 25px;
}

.please-login {
  color: red;
  text-align: center;
}

