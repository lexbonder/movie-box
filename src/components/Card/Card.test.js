import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

describe('Card', () => {
  let mockMovieArray;
  let mockUser;
  let mockMovie;

  beforeEach(() => {
    mockMovieArray = [
      {
        title: "The Maze Runner",
        vote_average: 7,
        release_date: "2014",
        overview: "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
        poster_path: "https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
        movie_id: 198663,
        favorite: false
      },
      {
        title: "Minions",
        vote_average: 6.4,
        release_date: "2015",
        overview: "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
        poster_path: "https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg",
        movie_id: 211672,
        favorite: false
      }
    ]
    mockUser = {id: 1, name:'Bob'}
    mockMovie = {
      title: "Minions",
      vote_average: 6.4,
      release_date: "2015",
      overview: "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
      poster_path: "https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg",
      movie_id: 211672,
      favorite: false
    }
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<Card movies={mockMovieArray} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should have default state', () => {
    const wrapper = shallow(<Card movies={mockMovieArray} />)
    const expectedState = {favWithoutUser: false}

    expect(wrapper.state()).toEqual(expectedState)
  })

  describe('handleFavClick', () => {
    it('should call toggle favorite if a user is logged in', () => {
      const mockEvent = {preventDefault: jest.fn()}
      const wrapper = shallow(<Card movies={mockMovieArray} user={mockUser} />)
      wrapper.instance().toggleFavorite = jest.fn()

      wrapper.instance().handleFavClick(mockEvent, mockMovie)

      expect(wrapper.instance().toggleFavorite).toHaveBeenCalled()
    })

    it('should change the state if a user is not logged in', () => {
      const mockEvent = {preventDefault: jest.fn()}
      const wrapper = shallow(<Card movies={mockMovieArray} user={{undefined}}/>)

      wrapper.instance().handleFavClick(mockEvent, mockMovie)

      expect(wrapper.state().favWithoutUser).toEqual(true)
    })
  })
})