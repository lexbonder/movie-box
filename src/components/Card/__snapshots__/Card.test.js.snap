// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movies={
      Array [
        Object {
          "favorite": false,
          "movie_id": 198663,
          "overview": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
          "poster_path": "https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
          "release_date": "2014",
          "title": "The Maze Runner",
          "vote_average": 7,
        },
        Object {
          "favorite": false,
          "movie_id": 211672,
          "overview": "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
          "poster_path": "https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg",
          "release_date": "2015",
          "title": "Minions",
          "vote_average": 6.4,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Card flip-container"
        >
          <div
            className="posterWrapper flipper"
          >
            <button
              className="front"
            >
              ★
            </button>
            <button
              className="back"
              id={undefined}
              onClick={[Function]}
            >
              ★
            </button>
            <img
              alt="Movie poster from The Maze Runner"
              className="front"
              src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
            />
            <div
              className="textBox back"
            >
              <h1
                className="movie-title"
              >
                The Maze Runner
              </h1>
              <p
                className="date"
              >
                2014
              </p>
              <p
                className="rating"
              >
                Rating: 7 / 10
              </p>
              <p>
                Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
              </p>
            </div>
          </div>
        </div>,
        <div
          className="Card flip-container"
        >
          <div
            className="posterWrapper flipper"
          >
            <button
              className="front"
            >
              ★
            </button>
            <button
              className="back"
              id={undefined}
              onClick={[Function]}
            >
              ★
            </button>
            <img
              alt="Movie poster from Minions"
              className="front"
              src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
            />
            <div
              className="textBox back"
            >
              <h1
                className="movie-title"
              >
                Minions
              </h1>
              <p
                className="date"
              >
                2015
              </p>
              <p
                className="rating"
              >
                Rating: 6.4 / 10
              </p>
              <p>
                Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
              </p>
            </div>
          </div>
        </div>,
      ],
      "className": "cardWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="posterWrapper flipper"
          >
            <button
              className="front"
            >
              ★
            </button>
            <button
              className="back"
              id={undefined}
              onClick={[Function]}
            >
              ★
            </button>
            <img
              alt="Movie poster from The Maze Runner"
              className="front"
              src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
            />
            <div
              className="textBox back"
            >
              <h1
                className="movie-title"
              >
                The Maze Runner
              </h1>
              <p
                className="date"
              >
                2014
              </p>
              <p
                className="rating"
              >
                Rating: 7 / 10
              </p>
              <p>
                Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
              </p>
            </div>
          </div>,
          "className": "Card flip-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="front"
              >
                ★
              </button>,
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>,
              <img
                alt="Movie poster from The Maze Runner"
                className="front"
                src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
              />,
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  The Maze Runner
                </h1>
                <p
                  className="date"
                >
                  2014
                </p>
                <p
                  className="rating"
                >
                  Rating: 7 / 10
                </p>
                <p>
                  Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                </p>
              </div>,
            ],
            "className": "posterWrapper flipper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "★",
                "className": "front",
              },
              "ref": null,
              "rendered": "★",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "★",
                "className": "back",
                "id": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "★",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Movie poster from The Maze Runner",
                "className": "front",
                "src": "https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="movie-title"
                  >
                    The Maze Runner
                  </h1>,
                  <p
                    className="date"
                  >
                    2014
                  </p>,
                  <p
                    className="rating"
                  >
                    Rating: 7 / 10
                  </p>,
                  <p>
                    Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                  </p>,
                ],
                "className": "textBox back",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The Maze Runner",
                    "className": "movie-title",
                  },
                  "ref": null,
                  "rendered": "The Maze Runner",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2014",
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": "2014",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Rating: 7 / 10",
                    "className": "rating",
                  },
                  "ref": null,
                  "rendered": "Rating: 7 / 10",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
                  },
                  "ref": null,
                  "rendered": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="posterWrapper flipper"
          >
            <button
              className="front"
            >
              ★
            </button>
            <button
              className="back"
              id={undefined}
              onClick={[Function]}
            >
              ★
            </button>
            <img
              alt="Movie poster from Minions"
              className="front"
              src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
            />
            <div
              className="textBox back"
            >
              <h1
                className="movie-title"
              >
                Minions
              </h1>
              <p
                className="date"
              >
                2015
              </p>
              <p
                className="rating"
              >
                Rating: 6.4 / 10
              </p>
              <p>
                Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
              </p>
            </div>
          </div>,
          "className": "Card flip-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="front"
              >
                ★
              </button>,
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>,
              <img
                alt="Movie poster from Minions"
                className="front"
                src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
              />,
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  Minions
                </h1>
                <p
                  className="date"
                >
                  2015
                </p>
                <p
                  className="rating"
                >
                  Rating: 6.4 / 10
                </p>
                <p>
                  Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                </p>
              </div>,
            ],
            "className": "posterWrapper flipper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "★",
                "className": "front",
              },
              "ref": null,
              "rendered": "★",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "★",
                "className": "back",
                "id": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "★",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Movie poster from Minions",
                "className": "front",
                "src": "https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="movie-title"
                  >
                    Minions
                  </h1>,
                  <p
                    className="date"
                  >
                    2015
                  </p>,
                  <p
                    className="rating"
                  >
                    Rating: 6.4 / 10
                  </p>,
                  <p>
                    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                  </p>,
                ],
                "className": "textBox back",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Minions",
                    "className": "movie-title",
                  },
                  "ref": null,
                  "rendered": "Minions",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2015",
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": "2015",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Rating: 6.4 / 10",
                    "className": "rating",
                  },
                  "ref": null,
                  "rendered": "Rating: 6.4 / 10",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
                  },
                  "ref": null,
                  "rendered": "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Card flip-container"
          >
            <div
              className="posterWrapper flipper"
            >
              <button
                className="front"
              >
                ★
              </button>
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>
              <img
                alt="Movie poster from The Maze Runner"
                className="front"
                src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
              />
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  The Maze Runner
                </h1>
                <p
                  className="date"
                >
                  2014
                </p>
                <p
                  className="rating"
                >
                  Rating: 7 / 10
                </p>
                <p>
                  Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                </p>
              </div>
            </div>
          </div>,
          <div
            className="Card flip-container"
          >
            <div
              className="posterWrapper flipper"
            >
              <button
                className="front"
              >
                ★
              </button>
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>
              <img
                alt="Movie poster from Minions"
                className="front"
                src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
              />
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  Minions
                </h1>
                <p
                  className="date"
                >
                  2015
                </p>
                <p
                  className="rating"
                >
                  Rating: 6.4 / 10
                </p>
                <p>
                  Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                </p>
              </div>
            </div>
          </div>,
        ],
        "className": "cardWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="posterWrapper flipper"
            >
              <button
                className="front"
              >
                ★
              </button>
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>
              <img
                alt="Movie poster from The Maze Runner"
                className="front"
                src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
              />
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  The Maze Runner
                </h1>
                <p
                  className="date"
                >
                  2014
                </p>
                <p
                  className="rating"
                >
                  Rating: 7 / 10
                </p>
                <p>
                  Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                </p>
              </div>
            </div>,
            "className": "Card flip-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="front"
                >
                  ★
                </button>,
                <button
                  className="back"
                  id={undefined}
                  onClick={[Function]}
                >
                  ★
                </button>,
                <img
                  alt="Movie poster from The Maze Runner"
                  className="front"
                  src="https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg"
                />,
                <div
                  className="textBox back"
                >
                  <h1
                    className="movie-title"
                  >
                    The Maze Runner
                  </h1>
                  <p
                    className="date"
                  >
                    2014
                  </p>
                  <p
                    className="rating"
                  >
                    Rating: 7 / 10
                  </p>
                  <p>
                    Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                  </p>
                </div>,
              ],
              "className": "posterWrapper flipper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "★",
                  "className": "front",
                },
                "ref": null,
                "rendered": "★",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "★",
                  "className": "back",
                  "id": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "★",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Movie poster from The Maze Runner",
                  "className": "front",
                  "src": "https://image.tmdb.org/t/p/w500/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="movie-title"
                    >
                      The Maze Runner
                    </h1>,
                    <p
                      className="date"
                    >
                      2014
                    </p>,
                    <p
                      className="rating"
                    >
                      Rating: 7 / 10
                    </p>,
                    <p>
                      Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.
                    </p>,
                  ],
                  "className": "textBox back",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "The Maze Runner",
                      "className": "movie-title",
                    },
                    "ref": null,
                    "rendered": "The Maze Runner",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2014",
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": "2014",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Rating: 7 / 10",
                      "className": "rating",
                    },
                    "ref": null,
                    "rendered": "Rating: 7 / 10",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
                    },
                    "ref": null,
                    "rendered": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="posterWrapper flipper"
            >
              <button
                className="front"
              >
                ★
              </button>
              <button
                className="back"
                id={undefined}
                onClick={[Function]}
              >
                ★
              </button>
              <img
                alt="Movie poster from Minions"
                className="front"
                src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
              />
              <div
                className="textBox back"
              >
                <h1
                  className="movie-title"
                >
                  Minions
                </h1>
                <p
                  className="date"
                >
                  2015
                </p>
                <p
                  className="rating"
                >
                  Rating: 6.4 / 10
                </p>
                <p>
                  Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                </p>
              </div>
            </div>,
            "className": "Card flip-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="front"
                >
                  ★
                </button>,
                <button
                  className="back"
                  id={undefined}
                  onClick={[Function]}
                >
                  ★
                </button>,
                <img
                  alt="Movie poster from Minions"
                  className="front"
                  src="https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg"
                />,
                <div
                  className="textBox back"
                >
                  <h1
                    className="movie-title"
                  >
                    Minions
                  </h1>
                  <p
                    className="date"
                  >
                    2015
                  </p>
                  <p
                    className="rating"
                  >
                    Rating: 6.4 / 10
                  </p>
                  <p>
                    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                  </p>
                </div>,
              ],
              "className": "posterWrapper flipper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "★",
                  "className": "front",
                },
                "ref": null,
                "rendered": "★",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "★",
                  "className": "back",
                  "id": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "★",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Movie poster from Minions",
                  "className": "front",
                  "src": "https://image.tmdb.org/t/p/w500/q0R4crx2SehcEEQEkYObktdeFy.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="movie-title"
                    >
                      Minions
                    </h1>,
                    <p
                      className="date"
                    >
                      2015
                    </p>,
                    <p
                      className="rating"
                    >
                      Rating: 6.4 / 10
                    </p>,
                    <p>
                      Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
                    </p>,
                  ],
                  "className": "textBox back",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Minions",
                      "className": "movie-title",
                    },
                    "ref": null,
                    "rendered": "Minions",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2015",
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": "2015",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Rating: 6.4 / 10",
                      "className": "rating",
                    },
                    "ref": null,
                    "rendered": "Rating: 6.4 / 10",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
                    },
                    "ref": null,
                    "rendered": "Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
